### PR TITLE
Feature abstract get type

### PR DESCRIPTION
**Pull request by eclipxe13**
Move concrete implementation of `getType` from `AbstractCollection` to `Collection`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ build
 composer.lock
 phpunit.xml
 vendor
+.idea
```
Comment (ramsey) [inline .gitignore]: Please make sure your editor adds a trailing newline character at the end of files.

**Comment by ramsey** [comment]
Looks good. Thanks for fixing my spelling mistakes. :-)

Add a trailing newline back to the `.gitignore` file, and I'll merge it in.

Thanks!

**Comment by eclipxe13** [comment]
done :)

**Comment by ramsey** [comment]
Thanks!

**Pull request merged**